Update Kubernetes Manifests to comment out liveness probe

keith-mcclellan opened the pull request:
Liveness probe can cause kubernetes to reboot nodes when the database is busy but functioning properly, so we are commenting out that probe.

keith-mcclellan committed:
Update Kubernetes Manifests to comment out liveness probe

Liveness probe can cause kubernetes to reboot nodes when the database is busy but functioning properly, so we are commenting out that probe.

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=67080) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=67080) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=67080) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67080)
<!-- Reviewable:end -->

taroface reviewed (approved):
Just a phrasing suggestion!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67080#-:-MdT2EKfDBnuKcEu1LsL:bw6tym9)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @chrisseto and @keith-mcclellan)

___
*[cloud/kubernetes/cockroachdb-statefulset.yaml, line 118 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/67080#-MdT1aUI7XWRvXzlicMo:-MdT1aUI7XWRvXzlicMp:b-mncmca) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ce02ca9deae60901d62c36d1f46b7b85026b058/cloud/kubernetes/cockroachdb-statefulset.yaml#L118)):*
> ```YAML
>         - containerPort: 8080
>           name: http
> # We recommend that users don't configure liveness probes for production environments as it can impact availability of production databases.
> ```

I suggest this wording in all the manifests:

`We recommend that you do not configure a liveness probe on a production environment, as this can impact the availability of production databases.`

Do we want to add something like "A liveness probe can be used to verify the health of a test deployment." (if that is true)

I also suggest indenting this at the same level as `livenessProbe` to be consistent with the other comments.


<!-- Sent from Reviewable.io -->

keith-mcclellan commented:
> Just a phrasing suggestion!
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67080#-:-MdT2EKfDBnuKcEu1LsL:bw6tym9)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @chrisseto and @keith-mcclellan)
> 
> _[cloud/kubernetes/cockroachdb-statefulset.yaml, line 118 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/67080#-MdT1aUI7XWRvXzlicMo:-MdT1aUI7XWRvXzlicMp:b-mncmca) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ce02ca9deae60901d62c36d1f46b7b85026b058/cloud/kubernetes/cockroachdb-statefulset.yaml#L118)):_
> 
> > ```yaml
> >         - containerPort: 8080
> >           name: http
> > # We recommend that users don't configure liveness probes for production environments as it can impact availability of production databases.
> > ```
> 
> I suggest this wording in all the manifests:
> 
> `We recommend that you do not configure a liveness probe on a production environment, as this can impact the availability of production databases.`
> 
> Do we want to add something like "A liveness probe can be used to verify the health of a test deployment." (if that is true)
> 
> I also suggest indenting this at the same level as `livenessProbe` to be consistent with the other comments.

Happy to change the wording.  The second one isn't really true - liveness probes are good for "quiet" environments but noisy ones they're bad.  This is something we'd probably want to expound on in the docs rather than a yaml comment.  What do you think @taroface ?

taroface commented:
Sounds good. I'll create a separate ticket to document the use of `livenessProbe` in e.g. the K8s monitoring doc.

keith-mcclellan committed:
change formatting and update comment

changed content description and updated commented out format to be in line with the rest of the yaml comments

keith-mcclellan commented:
bors r+

taroface commented:
bors r+
